pkg/infra/iac2/templates/eks_node_group/factory.ts

```diff
@@ -19,11 +19,11 @@ interface Args {
 // noinspection JSUnusedLocalSymbols
 function create(args: Args): aws.eks.NodeGroup {
     return new aws.eks.NodeGroup(args.Name, {
-        clusterName: args.Cluster.name,
+        clusterName: args.Cluster.name.apply((n) => n!),
```

Comment:
why do you need to run apply here? I thought input can be pulumi output

Reply:
`clusterName: pulumi.Input<string>`
whereas
`args.Cluster.name: pulumi.Output<string | undefined>`
😢

Reply:
i wonder why it can be undefined according to pulumi, but 🤷